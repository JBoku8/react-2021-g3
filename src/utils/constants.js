export const AUTH_TOKEN = 'auth.token';
export const TEXTS_STORAGE_KEY = 'TEXTS_STORAGE_KEY';
