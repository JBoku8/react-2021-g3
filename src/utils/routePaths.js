export const NOTES_PATH = '/notes';
export const COUNTER_PATH = '/counter';
export const PROFILE_PATH = '/profile';
export const FAKER_TEXT_PATH = '/faker-texts';
export const REDUX_COUNTER_PATH = '/redux-counter';
export const HOME_PATH = '/';

// auth
export const AUTH_PATH = '/auth';
export const SIGN_IN_PATH = '/auth/sign-in';
export const SIGN_UP_PATH = '/auth/sign-up';
