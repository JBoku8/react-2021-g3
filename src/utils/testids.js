export const TEST_IDS = {
  greeting: {
    title: 'greeting.title',
  },
  loader: {
    message: 'loader.message',
  },
  button: 'button',
};
