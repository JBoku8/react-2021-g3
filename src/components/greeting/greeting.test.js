import { render } from '@testing-library/react';
import { TEST_IDS } from '../../utils/testids';
import Greeting from './greeting';

describe('Greeting component test', () => {
  it('should render default title', () => {
    const component = render(<Greeting />);
    const titleElement = component.getByTestId(TEST_IDS.greeting.title);

    // get
    // query
    // find

    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent('');
  });

  it('should render with correct title props', () => {
    const mockProps = {
      text: 'mock text',
    };

    const component = render(<Greeting {...mockProps} />);
    const titleElement = component.getByTestId(TEST_IDS.greeting.title);

    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent(mockProps.text);
    expect(titleElement).toHaveAttribute('title', mockProps.text);
  });
});

describe('Greeting snapshots', () => {
  it('renders successfully snap 1', () => {
    const component = render(<Greeting />);

    expect(component).toMatchSnapshot();
  });

  it('renders successfully snap 2', () => {
    const mockProps = {
      text: 'mock text',
    };
    const component = render(<Greeting {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
