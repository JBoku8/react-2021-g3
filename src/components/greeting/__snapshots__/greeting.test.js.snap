// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting snapshots renders successfully snap 1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <h1
          class="text-muted text-dark"
          data-testid="greeting.title"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <h1
        class="text-muted text-dark"
        data-testid="greeting.title"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Greeting snapshots renders successfully snap 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <h1
          class="text-muted text-dark"
          data-testid="greeting.title"
          title="mock text"
        >
          mock text
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <h1
        class="text-muted text-dark"
        data-testid="greeting.title"
        title="mock text"
      >
        mock text
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
